framework/src/meshlet_helpers.cpp

```diff
@@ -0,0 +1,139 @@
+﻿#include <gvk.hpp>
+
+namespace gvk
+{
+	std::vector<meshlet> divide_into_meshlets(const std::vector<std::tuple<avk::resource_ownership<model_t>, std::vector<mesh_index_t>>>& aModels,
+		const uint32_t aMaxVertices, const uint32_t aMaxIndices, const bool aCombineSubmeshes)
+	{
+		return divide_into_meshlets(aModels, divide_into_very_bad_meshlets, aMaxVertices, aMaxIndices, aCombineSubmeshes);
+	}
+
+	std::vector<meshlet> divide_into_meshlets_cached(gvk::serializer& aSerializer,
+		const std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aModels,
+		const uint32_t aMaxVertices, const uint32_t aMaxIndices, const bool aCombineSubmeshes)
+	{
+		return divide_into_meshlets_cached(aSerializer, aModels, divide_into_very_bad_meshlets, aMaxVertices, aMaxIndices, aCombineSubmeshes);
+	}
+
+	template <typename F>
+	std::vector<meshlet> divide_into_meshlets(const std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aModels, F&& aMeshletDivision,
+		const uint32_t aMaxVertices, const uint32_t aMaxIndices, const bool aCombineSubmeshes)
+	{
+		std::vector<meshlet> meshlets;
+		for (auto& pair : aModels) {
+			auto model = std::get<avk::resource_ownership<model_t>>(pair);
+			auto& meshIndices = std::get<std::vector<mesh_index_t>>(pair);
+
+			if (aCombineSubmeshes) {
+				auto [vertices, indices] = get_vertices_and_indices(std::vector({ std::make_tuple(avk::const_referenced(model.get()), meshIndices) }));
+				// default to very bad meshlet generation
+				std::vector<meshlet> tmpMeshlets = divide_vertices_into_meshlets(vertices, indices, model, std::nullopt, aMaxVertices, aMaxIndices, aMeshletDivision);
```

Comment:
```suggestion
				std::vector<meshlet> tmpMeshlets = divide_vertices_into_meshlets(vertices, indices, std::move(model), std::nullopt, aMaxVertices, aMaxIndices, aMeshletDivision);
```

I think, this should make sense now. (Untested, though)

Reply:
model is already of type `const resource_ownership<model_t>&` here, move would not do anything if I'm not wrong?

Reply:
The old code was like this:
```
auto model = std::get<avk::resource_ownership<model_t>>(pair);
```
This performs definitely a copy of `avk::resource_ownership<model_t>`, regardless of what `std::get` returns!     
`auto` does _not_ add reference qualifiers automatically! Please see here: https://stackoverflow.com/questions/11886930/auto-specifier-type-deduction-for-references

Pointers are added automatically (otherwise they couldn't be assigned), but not references.

Reply:
Yes true, but I was talking about the const modifier which is deduced by auto if it's a reference. https://stackoverflow.com/a/32130883
I'll resolve this conversation now as it is already handled further down the line.